wrappers/hnynethttp/nethttp.go

```diff
@@ -69,29 +60,88 @@ func WrapHandler(handler http.Handler) http.Handler {
 func WrapHandlerFunc(hf func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
 	handlerFuncName := runtime.FuncForPC(reflect.ValueOf(hf).Pointer()).Name()
 	return func(w http.ResponseWriter, r *http.Request) {
-		start := time.Now()
-		ev := beeline.ContextEvent(r.Context())
-		if ev == nil {
-			ev = libhoney.NewEvent()
-			// put the event on the context for everybody downstream to use
-			r = r.WithContext(beeline.ContextWithEvent(r.Context(), ev))
-		}
-		// add some common fields from the request to our event
-		internal.AddRequestProps(r, ev)
+		ctx, span := common.StartSpanOrTraceFromHTTP(r)
+		defer span.Finish()
+		// push the context with our trace and span on to the request
+		r = r.WithContext(ctx)
 		// replace the writer with our wrapper to catch the status code
-		wrappedWriter := internal.NewResponseWriter(w)
+		wrappedWriter := common.NewResponseWriter(w)
 		// add the name of the handler func we're about to invoke
 		if handlerFuncName != "" {
-			ev.AddField("handler_func_name", handlerFuncName)
-			ev.AddField("name", handlerFuncName)
+			span.AddField("handler_func_name", handlerFuncName)
+			span.AddField("name", handlerFuncName)
 		}
 
 		hf(wrappedWriter, r)
 		if wrappedWriter.Status == 0 {
 			wrappedWriter.Status = 200
 		}
-		ev.AddField("response.status_code", wrappedWriter.Status)
-		ev.AddField("duration_ms", float64(time.Since(start))/float64(time.Millisecond))
-		ev.Send()
+		span.AddField("response.status_code", wrappedWriter.Status)
+	}
+}
+
+type hnyTripper struct {
+	// wrt is the wrapped round tripper
+	wrt http.RoundTripper
+}
+
+func (ht *hnyTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	ctx := r.Context()
+	span := trace.GetSpanFromContext(ctx)
+	if span == nil {
+		// if there's no trace in the context, just send an event
+		tm := timer.Start()
+		ev := libhoney.NewEvent()
+		defer ev.Send()
+
+		// add in common request headers.
+		for k, v := range common.GetRequestProps(r) {
+			span.AddField(k, v)
+		}
+
+		ev.AddField("meta.type", "http_client")
+
+		resp, err := ht.wrt.RoundTrip(r)
+
+		if err != nil {
+			// TODO should this error field be namespaced somehow
+			ev.AddField("error", err.Error())
+		}
+		dur := tm.Finish()
+		ev.AddField("duration_ms", dur)
+		return resp, err
+	}
+	// we have a trace, let's use it and pass along trace context in addition to
+	// making a span around this HTTP call
+	ctx, span = span.ChildSpan(ctx)
+	defer span.Finish()
+	r = r.WithContext(ctx)
+	// add in common request headers.
+	for k, v := range common.GetRequestProps(r) {
+		span.AddField(k, v)
+	}
+	span.AddField("meta.type", "http_client")
```

Comment:
yeah, there is, but it's only a few lines on either side of the call to the wrapped RoundTripper. Too few to warrant another function in the same vein as with the db wrapping.  hrmph. Going to leave it but separate them into functions rather than a large if/else clause.